src/python/pants/backend/jvm/tasks/classpath_entry.py

```diff
@@ -24,6 +24,9 @@ def path(self):
     """
     return self._path
 
+  def set_directory_digest(self, directory_digest):
+    self._directory_digest = directory_digest
```

Comment:
I was thinking whether we should make less mutable like this:
```
assert self._directory_digest is None, 'Mutating non-None directory_digest is not allowed'
self._directory_digest = directory_digest
```

Reply:
That sounds great! And might be even all of the "type safety" we need for now! Will do!

Reply:
Done!